contracts/GameJam.sol

```diff
@@ -23,4 +54,26 @@ contract GameJam {
 
     emit CompetitorAdded(competitor);
   }
+  // finish function is a payable used to declare the winner via their address
+  // the function should only be called able when the GameJam is in Progress
+  // otherwise, it would be possible to finish again and again
+  function finish(address winner) public payable
+    onlyAtStage(Stages.JamInProgress)
+    transitionNext {
+
+
+      emit WinnerDeclared(winner);
+    }
+
+  // payoutWinner is used after the Jam to payout the declared winner
+  // and ultimately finish the GameJam.
+  function payoutWinner(address payable winner) public payable
+    onlyAtStage(Stages.WinnerDeclaration)
+    transitionNext {
+      // If there are prizes to distribute
+      if (balance > 0) {
```

Comment:
Should we set the balance to 0 after this?

Reply:
Now setting balance to zero after transfer of winnings.